modules/core/src/alloc.cpp

```diff
@@ -100,25 +100,24 @@ static bool readMemoryAlignmentParameter()
     // TODO add checks for valgrind, ASAN if value == false
     return value;
 }
+
+#if defined _MSC_VER
+#pragma warning(suppress:4714)  // preventive: const marked as __forceinline not inlined
+static __forceinline
+#else
 static inline
+#endif
 bool isAlignedAllocationEnabled()
 {
-    static bool initialized = false;
-    static bool useMemalign = true;
-    if (!initialized)
-    {
-        initialized = true;  // trick to avoid stuck in acquire (works only if allocations are scope based)
-        useMemalign = readMemoryAlignmentParameter();
-    }
+    // use construct on first use idiom https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use
+    // details: https://github.com/opencv/opencv/issues/15691
+    static bool useMemalign
+    #if defined __GNUC__
+        __attribute__((unused))
+    #endif
```

Comment:
not needed

Reply:
OK. I'll make the change and spin up tests Monday.